openvpn/middlewares/server/auth/middleware_test.go

```diff
@@ -0,0 +1,191 @@
+		assert.NoError(t, err, test.line)
+		assert.False(t, consumed, test.line)
+	}
+}
+
+func Test_ConsumeLineTakes(t *testing.T) {
+	var tests = []struct {
+		line string
+	}{
+		{">CLIENT:REAUTH,0,0"},
+		{">CLIENT:CONNECT,0,0"},
+		{">CLIENT:ENV,password=12341234"},
+		{">CLIENT:ENV,username=username"},
+	}
+
+	middleware := NewMiddleware(fakeAuthenticator)
+	connection := &fakeConnection{}
+	middleware.Start(connection)
+
+	for _, test := range tests {
+		consumed, err := middleware.ConsumeLine(test.line)
+		assert.NoError(t, err, test.line)
+		assert.True(t, consumed, test.line)
+	}
+}
+
+func Test_ConsumeLineAuthState(t *testing.T) {
+	var tests = []struct {
+		line          string
+		expectedState openvpn.State
+	}{
+		{">CLIENT:REAUTH,0,0", openvpn.STATE_AUTH},
+		{">CLIENT:CONNECT,0,0", openvpn.STATE_AUTH},
+	}
+
+	for _, test := range tests {
+		middleware := NewMiddleware(fakeAuthenticator)
+		connection := &fakeConnection{}
+		middleware.Start(connection)
+
+		consumed, err := middleware.ConsumeLine(test.line)
+		assert.NoError(t, err, test.line)
+		assert.True(t, consumed, test.line)
+		assert.Equal(t, test.expectedState, middleware.state)
+	}
+}
+
+func Test_ConsumeLineNotAuthState(t *testing.T) {
+	var tests = []struct {
+		line            string
+		unexpectedState openvpn.State
+	}{
+		{">CLIENT:ENV,password=12341234", openvpn.STATE_AUTH},
+		{">CLIENT:ENV,username=username", openvpn.STATE_AUTH},
+	}
+
+	for _, test := range tests {
+		middleware := NewMiddleware(fakeAuthenticator)
+		connection := &fakeConnection{}
+		middleware.Start(connection)
+
+		consumed, err := middleware.ConsumeLine(test.line)
+		assert.NoError(t, err, test.line)
+		assert.False(t, consumed, test.line)
+		assert.NotEqual(t, test.unexpectedState, middleware.state)
+	}
+}
+
+func Test_ConsumeLineParseChecker(t *testing.T) {
+	var tests = []struct {
+		line          string
+		expectedState openvpn.State
+	}{
+		{">CLIENT:CONNECT,0,0", openvpn.STATE_AUTH},
+		{">CLIENT:ENV,password=12341234", openvpn.STATE_AUTH},
+		{">CLIENT:ENV,username=username1", openvpn.STATE_AUTH},
+		{">CLIENT:ENV,END", openvpn.STATE_UNDEFINED},
+	}
+
+	middleware := NewMiddleware(fakeAuthenticator)
+	connection := &fakeConnection{}
+	middleware.Start(connection)
+
+	for _, test := range tests {
+		consumed, err := middleware.ConsumeLine(test.line)
+		assert.NoError(t, err, test.line)
+		assert.True(t, consumed, test.line)
+		assert.Equal(t, test.expectedState, middleware.state)
+	}
+	assert.Equal(t, "username1", middleware.lastUsername)
+	assert.Equal(t, "12341234", middleware.lastPassword)
+}
+
+func Test_ConsumeLineAuthTrueChecker(t *testing.T) {
+	var tests = []struct {
+		line          string
+		expectedState openvpn.State
+	}{
+		{">CLIENT:CONNECT,0,0", openvpn.STATE_AUTH},
+		{">CLIENT:ENV,password=12341234", openvpn.STATE_AUTH},
+		{">CLIENT:ENV,username=username1", openvpn.STATE_AUTH},
+		{">CLIENT:ENV,END", openvpn.STATE_UNDEFINED},
+	}
+	authFake := NewAuthenticatorFake()
```

Comment:
Construct local callback function and You will be able to assert if `authFake` was called